### PR TITLE
stable diffusion image to image cli and sdk jobs

### PR DESCRIPTION
**Pull request by vadthyavath**
# Description

# Checklist

- [x] I have read the contribution guidelines.
  - [General](https://github.com/Azure/azureml-examples/blob/main/CONTRIBUTING.md)
  - [SDK](https://github.com/Azure/azureml-examples/blob/main/sdk/CONTRIBUTING.md)
  - [CLI](https://github.com/Azure/azureml-examples/blob/main/cli/CONTRIBUTING.md)
- [x] I have coordinated with the docs team (mldocs@microsoft.com) if this PR deletes files or changes any file names or file extensions.
- [ ] Pull request includes test coverage for the included changes.
- [ ] This notebook or file is added to the [CODEOWNERS](https://github.com/Azure/azureml-examples/blob/main/.github/CODEOWNERS) file, pointing to the author or the author's team.

### CODE REVIEW

File: sdk/python/foundation-models/system/inference/text-to-image/image-text-to-image-online-endpoint.ipynb [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+    "    display(img)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "generations"
```
Comment (MadhuM02) [inline sdk/python/foundation-models/system/inference/text-to-image/image-text-to-image-online-endpoint.ipynb:300]: can this be removed??

File: cli/foundation-models/system/inference/text-to-image/batch-deploy.yml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 $schema: https://azuremlschemas.azureedge.net/latest/batchDeployment.schema.json
-description: "Batch endpoint for for text-to-image task"
+description: "Batch endpoint for for image-text-to-image task"
```
Comment (rjaincc) [inline cli/foundation-models/system/inference/text-to-image/batch-deploy.yml:2]: This is a common file. 

```suggestion
description: "Batch endpoint for for (image-)text-to-image task"
```

File: sdk/python/foundation-models/system/inference/text-to-image/image-text-to-image-batch-endpoint.ipynb [hunk 3]
```diff
@@ -0,0 +1,520 @@
...
+    "## Image to Image Inference using Online Endpoints\n",
+    "\n",
+    "This sample shows how to deploy `image to image` type stable diffusion models to a batch endpoint for inference.\n",
+    "\n",
+    "### Task\n",
+    "`image to image` task takes an original image and a text prompt as input. The model generates an image by modifying the original image.\n",
+    "\n",
+    " \n",
+    "### Model\n",
+    "Models that can perform the `image to image` task are tagged with `image-text-to-image`. We will use the `stabilityai-stable-diffusion-xl-refiner-1-0` model in this notebook. If you opened this notebook from a specific model card, remember to replace the specific model name.\n",
```
Comment (rjaincc) [inline sdk/python/foundation-models/system/inference/text-to-image/image-text-to-image-batch-endpoint.ipynb:17]: The task should be image to image. 
Have left the same comment on model asset PR - https://github.com/Azure/azureml-assets/pull/1785.

File: sdk/python/foundation-models/system/inference/text-to-image/image-text-to-image-batch-endpoint.ipynb [hunk 4]
```diff
@@ -0,0 +1,520 @@
...
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 2. Pick a model to deploy\n",
+    "\n",
+    "Browse models in the Model Catalog in the AzureML Studio, filtering by the `image-text-to-image` task. In this example, we use the `stabilityai-stable-diffusion-xl-refiner-1-0` model. If you have opened this notebook for a different model, replace the model name accordingly. This is a pre-trained model."
```
Comment (rjaincc) [inline sdk/python/foundation-models/system/inference/text-to-image/image-text-to-image-batch-endpoint.ipynb:127]: Task name. Same comment as above.

File: sdk/python/foundation-models/system/inference/text-to-image/image-text-to-image-online-endpoint.ipynb [hunk 5]
```diff
@@ -0,0 +1,342 @@
...
+    "## Image to Image Inference using Online Endpoints\n",
+    "\n",
+    "This sample shows how to deploy `image to image` type stable diffusion models to an online endpoint for inference.\n",
+    "\n",
+    "### Task\n",
+    "`image to image` task takes an original image and a text prompt as input. The model generates image by modifying the original image.\n",
+    "\n",
+    " \n",
+    "### Model\n",
+    "Models that can perform the `image to image` task are tagged with `image-text-to-image`. We will use the `stabilityai-stable-diffusion-xl-refiner-1-0` model in this notebook. If you opened this notebook from a specific model card, remember to replace the specific model name.\n",
```
Comment (rjaincc) [inline sdk/python/foundation-models/system/inference/text-to-image/image-text-to-image-online-endpoint.ipynb:16]: Task name. Same comment as above. 

The task should be image-to-image.
Have left the same comment on model asset PR - https://github.com/Azure/azureml-assets/pull/1785.

File: sdk/python/foundation-models/system/inference/text-to-image/image-text-to-image-online-endpoint.ipynb [hunk 6]
```diff
@@ -0,0 +1,342 @@
...
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "workspace_ml_client.online_endpoints.begin_delete(name=online_endpoint_name).wait()"
+   ]
+  }
+ ],
+ "metadata": {
```
Comment (rjaincc) [inline sdk/python/foundation-models/system/inference/text-to-image/image-text-to-image-online-endpoint.ipynb:321]: Clean up the metadata

File: sdk/python/foundation-models/system/inference/text-to-image/safe-image-text-to-image-batch-deployment.ipynb [hunk 7]
```diff
@@ -0,0 +1,786 @@
...
+    "\n",
+    "### Task\n",
+    "`image to image` task takes an original image, a text prompt as input. The model generates an image by modifying the original image.\n",
+    "\n",
+    " \n",
+    "### Model\n",
+    "Models that can perform the `image to image` task are tagged with `image-text-to-image`. We will use the `stabilityai-stable-diffusion-xl-refiner-1-0` model in this notebook. If you opened this notebook from a specific model card, remember to replace the specific model name.\n",
```
Comment (rjaincc) [inline sdk/python/foundation-models/system/inference/text-to-image/safe-image-text-to-image-batch-deployment.ipynb:20]: Task name. Same comment as above. 

The task should be image-to-image.
Have left the same comment on model asset PR - https://github.com/Azure/azureml-assets/pull/1785.

File: sdk/python/foundation-models/system/inference/text-to-image/safe-image-text-to-image-batch-deployment.ipynb [hunk 8]
```diff
@@ -0,0 +1,786 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 3. Create AACS enabled image to image batch endpoint\n",
+    "\n",
+    "#### 3.1 Check if image to image model is available in the AML registry\n",
+    "\n",
+    "Browse models in the Model Catalog in the AzureML Studio, filtering by the image-text-to-image task. In this example, we use the `stabilityai-stable-diffusion-xl-refiner-1-0 model`. If you have opened this notebook for a different model, replace the model name accordingly. This is a pre-trained model."
```
Comment (rjaincc) [inline sdk/python/foundation-models/system/inference/text-to-image/safe-image-text-to-image-batch-deployment.ipynb:265]: Task name. Same comment as above. 

The task should be image-to-image.
Have left the same comment on model asset PR - https://github.com/Azure/azureml-assets/pull/1785.

File: sdk/python/foundation-models/system/inference/text-to-image/image-text-to-image-online-endpoint.ipynb [hunk 9]
```diff
@@ -0,0 +1,342 @@
...
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 2. Pick a model to deploy\n",
+    "\n",
+    "Browse models in the Model Catalog in the AzureML Studio, filtering by the `image-text-to-image` task. In this example, we use the `stabilityai-stable-diffusion-xl-refiner-1-0` model. If you have opened this notebook for a different model, replace the model name accordingly. This is a pre-trained model."
```
Comment (rjaincc) [inline sdk/python/foundation-models/system/inference/text-to-image/image-text-to-image-online-endpoint.ipynb:83]: Task name. Same comment as above.

The task should be image-to-image.
Have left the same comment on model asset PR - https://github.com/Azure/azureml-assets/pull/1785.

File: sdk/python/foundation-models/system/inference/text-to-image/safe-image-text-to-image-online-endpoint.ipynb [hunk 10]
```diff
@@ -0,0 +1,761 @@
...
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Create an Azure AI Content Safety __(AACS)__ enabled Image to Image online endpoint (Preview)\n",
+    "### This notebook is under preview.\n",
+    "\n",
+    "### Steps to create an __AACS__ enabled __image-to-image__ online endpoint\n",
+    "1. Create an __AACS__ resource for moderating the request from user and response from the __image-to-image__ online endpoint.\n",
+    "2. Create a new __AACS__ enabled __image-to-image__ online endpoint with a custom [score_online.py](./aacs-scoring-files/score/score_online.py) which will integrate with the __AACS__ resource to moderate the response from the __image-text-to-image__ model and the request from the user, but to make the custom [score_online.py](./aacs-scoring-files/score/score_online.py) to successfully authenticated to the __AACS__ resource, we have 2 options:\n",
```
Comment (rjaincc) [inline sdk/python/foundation-models/system/inference/text-to-image/safe-image-text-to-image-online-endpoint.ipynb:12]: Task name. Same comment as above.

The task should be image-to-image.
Have left the same comment on model asset PR - https://github.com/Azure/azureml-assets/pull/1785.

File: sdk/python/foundation-models/system/inference/text-to-image/safe-image-text-to-image-online-endpoint.ipynb [hunk 11]
```diff
@@ -0,0 +1,761 @@
...
+    "    2. __Environment variable__, simpler but less secure approach, is to just pass the access key of the AACS resource to the custom [score_online.py](./aacs-scoring-files/score/score_online.py) via environment variable, then the custom [score_online.py](./aacs-scoring-files/score/score_online.py) can use the key directly to access the AACS resource, this option is less secure than the first option, if someone in your org has access to the endpoint, he/she can get the access key from the environment variable and use it to access the AACS resource.\n",
+    "\n",
+    "\n",
+    "### Task\n",
+    "`image to image` task takes an original image, a text prompt as input. The model generates an image by modifying the original image.\n",
+    "\n",
+    " \n",
+    "### Model\n",
+    "Models that can perform the `image-to-image` task are tagged with `image-text-to-image`. We will use the `stabilityai-stable-diffusion-xl-refiner-1-0` model in this notebook. If you opened this notebook from a specific model card, remember to replace the specific model name.\n",
```
Comment (rjaincc) [inline sdk/python/foundation-models/system/inference/text-to-image/safe-image-text-to-image-online-endpoint.ipynb:22]: Task name. Same comment as above.

The task should be image-to-image.
Have left the same comment on model asset PR - https://github.com/Azure/azureml-assets/pull/1785.

File: sdk/python/foundation-models/system/inference/text-to-image/safe-image-text-to-image-online-endpoint.ipynb [hunk 12]
```diff
@@ -0,0 +1,761 @@
...
+    "        raise ex"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 3. Pick a model to deploy\n",
+    "\n",
+    "Browse models in the Model Catalog in the AzureML Studio, filtering by the `image-text-to-image` task. In this example, we use the `stabilityai-stable-diffusion-xl-refiner-1-0` model. If you have opened this notebook for a different model, replace the model name accordingly. This is a pre-trained model."
```
Comment (rjaincc) [inline sdk/python/foundation-models/system/inference/text-to-image/safe-image-text-to-image-online-endpoint.ipynb:301]: Task name. Same comment as above.

The task should be image-to-image.
Have left the same comment on model asset PR - https://github.com/Azure/azureml-assets/pull/1785.

**Review by rjaincc — DISMISSED** [review]
The task should be image-to-image.
Have left the same comment on model asset PR - https://github.com/Azure/azureml-assets/pull/1785.

**Review by gauravrajguru — APPROVED** [review]

**Review by rjaincc — APPROVED** [review]

**Pull request merged**